Completion.py

```diff
@@ -96,7 +139,16 @@ def notify_func(filepath, content, callback):
 
 def complete_func(filepath, row, col, content, error_cb, data_cb):
     settings = read_settings()
-    cli = http_client.YcmdClient(settings["server"], settings["port"], settings["hmac"])
```

Comment:
Maybe better solution is to move this logic of getting server,port,hmac to separate function and call in from all places.
Something like:

def get_ycmd_client(....):
     .....
     return http_client.YcmdClient(0, server, port, hmac)